src/Infrastructure/Persistence/Queries.hs

```diff
@@ -116,25 +115,32 @@ addContentWithTags content tags = transaction Serializable Write $ do
   Transaction.statement () $ add contentSchema [litContent content]
   Transaction.statement () $ add contentsTagsSchema (contentTag (litContent content) <$> (litTag <$> alreadyPresentTags) <> newTags)
 
--- SELECT USER BY USERNAME
+-- |
+-- Describes the possible error cases for queries that expect exactly one row as a result.
+data WrongNumberOfRows
```

Comment:
I would move this and the `singleUser` function below the `-- SELECT USER BY USERNAME` comment

Reply:
Hm, they're not specific to that function, that's why I moved it _out_. But if you think that's better, I'm ok with it.